src/utils/AddressUtils.ts

```diff
@@ -207,20 +203,18 @@ export class Address {
     }
   }
 
-  static isEvmAddress(address: Address): boolean {
-    return address instanceof EvmAddress;
+  isEVM(): this is EvmAddress {
```

Comment:
+1

Should we also make isAddress a type predicate?

isAddress is a bit tricker to avoid `instanceof`, but a potential workaround would be:
```ts
export abstract class Address {
  __address_type_brand = true;
...
  static isAddress(obj: unknown): obj is Address {
    return "__address_type_brand" in obj;
  }
...
}
```

Even though it's a bit nastier, I think it's safer.

Reply:
ooc, do you have a specific use case in mind for this? If I've understood correctly, it'd be most useful for narrowing from `unknown` or `any`. For `Address` types we typically pass them around as `Address` and have to subsequently narrow them down to the `EvmAddress` or `SvmAddress` variants, so I'm not sure where we'd actually need to use an `isAddress()` predicate.

Reply:
Dropped here: https://github.com/across-protocol/sdk/pull/1101/commits/5b6fd8cf393870057dca47751cd7b4ce054e477a